rs/bitcoin/adapter/tests/adapter_test.rs

```diff
@@ -503,6 +539,130 @@ fn test_receives_blocks() {
     assert_eq!(blocks.len(), 150);
 }
 
+// Checks that the adapter disconnects from the clients when it becomes idle.
+#[test]
+fn test_adapter_disconnects_when_idle() {
+    let logger = no_op_logger();
+
+    let bitcoind1 = get_default_bitcoind();
+    let client1 = Client::new(
+        bitcoind1.rpc_url().as_str(),
+        Auth::CookieFile(bitcoind1.params.cookie_file.clone()),
+    )
+    .unwrap();
+
+    let bitcoind2 = get_default_bitcoind();
+    let client2 = Client::new(
+        bitcoind2.rpc_url().as_str(),
+        Auth::CookieFile(bitcoind2.params.cookie_file.clone()),
+    )
+    .unwrap();
+
+    let bitcoind3 = get_default_bitcoind();
+    let client3 = Client::new(
+        bitcoind3.rpc_url().as_str(),
+        Auth::CookieFile(bitcoind3.params.cookie_file.clone()),
+    )
+    .unwrap();
+
+    let url1 = SocketAddr::V4(get_bitcoind_url(&bitcoind1).unwrap());
+    let url2 = SocketAddr::V4(get_bitcoind_url(&bitcoind2).unwrap());
+    let url3 = SocketAddr::V4(get_bitcoind_url(&bitcoind3).unwrap());
+
+    client1
+        .add_node(&url2.to_string())
+        .expect("Failed to connect to peer");
+    client2
+        .add_node(&url3.to_string())
+        .expect("Failed to connect to peer");
+    client3
+        .add_node(&url1.to_string())
+        .expect("Failed to connect to peer");
+
+    let rt: Runtime = tokio::runtime::Runtime::new().unwrap();
+
+    let _r = start_active_adapter_and_client(
+        &rt,
+        vec![url1, url2, url3],
+        logger,
+        bitcoin::Network::Regtest,
+    );
+
+    // Each client is connected to 3 peers: 2 peers from the other clients and 1 peer from the adapter.
+    wait_for_connection(&client1, 3);
+    wait_for_connection(&client2, 3);
+    wait_for_connection(&client3, 3);
+
+    // it takes 6 seconds for the adapter to become idle (in the test config).
+    std::thread::sleep(std::time::Duration::from_secs(7)); // wait for the adapter to become idle.
+
+    // As the adapter is now idle, the connection with the clients should be dropped.
+    // Hence each client should now be connected to just 2 peers.
+    exact_connections(&client1, 2);
+    exact_connections(&client2, 2);
+    exact_connections(&client3, 2);
+}
+
+/// Checks that an idle adapter does not connect to any peers.
+#[test]
+fn idle_adapter_does_not_connect_to_peers() {
+    let logger = no_op_logger();
+
+    let bitcoind1 = get_default_bitcoind();
```

Comment:
I don't think we need 3 bitcoin peers to test that the adapter doesn't connect when idle. I would have a single bitcoind in the test here